### ISSUE TITLE
Multiple CVEs that do not make any sense

### ISSUE THREAD

**Issue by hannes-angst** [IBM/cos-spring#1]
When I let OWASP dependency check run, it came back with some very nasty results:

|File | Vulnerability | Severity|CVSS|CWE|
| ---|--------------|---------|-----|----- |
|cos-spring-boot-starter-1.0.1.jar|CVE-2016-0264|Medium|6.8|CWE-119 Improper Restriction of Operations within the Bounds of a Memory Buffer|
|cos-spring-boot-starter-1.0.1.jar|CVE-2017-1289|Medium|6.4|CWE-611 Improper Restriction of XML External Entity Reference ('XXE')|
|cos-spring-boot-starter-1.0.1.jar|CVE-2015-5006|Low|2.1|CWE-200 Information Exposure|
|cos-spring-boot-starter-1.0.1.jar|CVE-2016-0363|Medium|6.8|CWE-20 Improper Input Validation|
|cos-spring-boot-starter-1.0.1.jar|CVE-2015-5041|Medium|6.4|CWE-200 Information Exposure|
|cos-spring-boot-starter-1.0.1.jar|CVE-2014-8891|High|10.0|-|
|cos-spring-boot-starter-1.0.1.jar|CVE-2016-3956|Medium|5.0|CWE-200 Information Exposure|
|cos-spring-boot-starter-1.0.1.jar|CVE-2014-8892|High|7.8|-|
|cos-spring-boot-starter-1.0.1.jar|CVE-2016-0376|Medium|5.1|-|
|ibm-cos-java-sdk-core-2.3.0.jar|CVE-2016-0376|Medium|5.1|-|
|ibm-cos-java-sdk-core-2.3.0.jar|CVE-2016-0264|Medium|6.8|CWE-119 Improper Restriction of Operations within the Bounds of a Memory Buffer|
|ibm-cos-java-sdk-core-2.3.0.jar|CVE-2015-5006|Low|2.1|CWE-200 Information Exposure|
|ibm-cos-java-sdk-core-2.3.0.jar|CVE-2016-0363|Medium|6.8|CWE-20 Improper Input Validation|
|ibm-cos-java-sdk-core-2.3.0.jar|CVE-2015-5041|Medium|6.4|CWE-200 Information Exposure|
|ibm-cos-java-sdk-core-2.3.0.jar|CVE-2014-8891|High|10.0|-|
|ibm-cos-java-sdk-core-2.3.0.jar|CVE-2014-8892|High|7.8|-|

However, as fare as I can tell, this CVEs are in no way related to COS *but* to a rather old version of IBM JRE. 

May you be so kind and file a false positive for your library.

**Comment by rtveitch** [comment]
Can you provide a bit more information about the tool and process you're following? I agree that these seem to be unrelated CVEs, if the below is what you're running it seems like the false positives can be ignored with some configuration of the tool:
https://jeremylong.github.io/DependencyCheck/general/suppression.html

**Comment by hannes-angst** [comment]
The table above is created using the owasp dependency checker plugin for jenkins, which in turn collects various findings from indices like the NVD from NIST. One way to correct the false positive is to suppress the findings by gav or CVE number, as you already pointed out. The other way would be to ask for a correction in the index itself. A maintainer of this codebase would probably be entitled to ask for a correction. But as I was never in such position I do not know how tedious this task might be.

**Comment by rtveitch** [comment]
@hannes-angst I'm having some trouble reproducing your results. Here are the steps I have followed:

1. clone the repo and cd into it
2. download all `cos-spring` dependencies and transitive dependencies using the following command:
```
mvn dependency:copy-dependencies -DoutputDirectory=/tmp/lib/
```
3. `ls` the output directory to validate that jars were downloaded, below is what I get:
```
ls /tmp/lib/
accessors-smart-1.2.jar                               jackson-annotations-2.9.0.jar                         spring-beans-5.0.6.RELEASE.jar
android-json-0.0.20131108.vaadin1.jar                 jackson-core-2.9.5.jar                                spring-boot-1.5.3.RELEASE.jar
asm-5.0.4.jar                                         jackson-databind-2.9.5.jar                            spring-boot-2.0.2.RELEASE.jar
assertj-core-3.9.1.jar                                jackson-dataformat-cbor-2.9.5.jar                     spring-boot-autoconfigure-1.5.3.RELEASE.jar
byte-buddy-1.7.9.jar                                  javax.annotation-api-1.3.2.jar                        spring-boot-autoconfigure-2.0.2.RELEASE.jar
byte-buddy-agent-1.7.9.jar                            joda-time-2.8.1.jar                                   spring-boot-configuration-processor-1.5.3.RELEASE.jar
commons-codec-1.9.jar                                 json-path-2.4.0.jar                                   spring-boot-configuration-processor-2.0.2.RELEASE.jar
commons-logging-1.1.3.jar                             json-smart-2.3.jar                                    spring-boot-starter-2.0.2.RELEASE.jar
commons-logging-1.2.jar                               jsonassert-1.5.0.jar                                  spring-boot-starter-logging-2.0.2.RELEASE.jar
cos-spring-boot-starter-1.0.1.jar                     jul-to-slf4j-1.7.25.jar                               spring-boot-starter-test-2.0.2.RELEASE.jar
cos-spring-framework-1.0.1.jar                        junit-4.12.jar                                        spring-boot-test-2.0.2.RELEASE.jar
hamcrest-core-1.3.jar                                 log4j-api-2.10.0.jar                                  spring-boot-test-autoconfigure-2.0.2.RELEASE.jar
hamcrest-library-1.3.jar                              log4j-to-slf4j-2.10.0.jar                             spring-context-4.3.17.RELEASE.jar
httpasyncclient-4.1.3.jar                             logback-classic-1.2.3.jar                             spring-context-5.0.6.RELEASE.jar
httpclient-4.5.2.jar                                  logback-core-1.2.3.jar                                spring-core-4.3.17.RELEASE.jar
httpcore-4.4.4.jar                                    mockito-core-2.15.0.jar                               spring-core-5.0.6.RELEASE.jar
httpcore-nio-4.4.6.jar                                objenesis-2.6.jar                                     spring-expression-4.3.17.RELEASE.jar
ibm-cos-java-sdk-2.3.0.jar                            slf4j-api-1.7.25.jar                                  spring-expression-5.0.6.RELEASE.jar
ibm-cos-java-sdk-core-2.3.0.jar                       snakeyaml-1.19.jar                                    spring-jcl-5.0.6.RELEASE.jar
ibm-cos-java-sdk-kms-2.3.0.jar                        spring-aop-4.3.17.RELEASE.jar                         spring-test-5.0.6.RELEASE.jar
ibm-cos-java-sdk-s3-2.3.0.jar                         spring-aop-5.0.6.RELEASE.jar                          xmlunit-core-2.5.1.jar
ion-java-1.0.2.jar                                    spring-beans-4.3.17.RELEASE.jar
```
4. download the owasp dependency checker cli from here and cd into it: https://www.owasp.org/index.php/OWASP_Dependency_Check
5. run the dependency checker as follows:
```
./bin/dependency-check.sh -s /tmp/lib/ --project cos-spring
```
Here is the summary from the top of the resulting report:

Dependency | Vulnerability IDs | Package | Highest Severity | CVE Count | Confidence | Evidence Count
-- | -- | -- | -- | -- | -- | --
spring-core-5.0.6.RELEASE.jar | cpe:2.3:a:pivotal_software:spring_framework:5.0.6.release:*:*:*:*:*:*:*cpe:2.3:a:springsource:spring_framework:5.0.6.release:*:*:*:*:*:*:*cpe:2.3:a:vmware:springsource_spring_framework:5.0.6:*:*:*:*:*:*:* | pkg:maven/org.springframework/spring-core@5.0.6.RELEASE | HIGH | 3 | Highest | 31
jackson-databind-2.9.5.jar | cpe:2.3:a:fasterxml:jackson:2.9.5:*:*:*:*:*:*:*cpe:2.3:a:fasterxml:jackson-databind:2.9.5:*:*:*:*:*:*:* | pkg:maven/com.fasterxml.jackson.core/jackson-databind@2.9.5 | CRITICAL | 13 | Highest | 40
spring-boot-1.5.3.RELEASE.jar | cpe:2.3:a:pivotal_software:spring_boot:1.5.3.release:*:*:*:*:*:*:* | pkg:maven/org.springframework.boot/spring-boot@1.5.3.RELEASE | CRITICAL | 2 | High | 30
spring-core-4.3.17.RELEASE.jar | cpe:2.3:a:pivotal_software:spring_framework:4.3.17.release:*:*:*:*:*:*:*cpe:2.3:a:springsource:spring_framework:4.3.17.release:*:*:*:*:*:*:*cpe:2.3:a:vmware:springsource_spring_framework:4.3.17:*:*:*:*:*:*:* | pkg:maven/org.springframework/spring-core@4.3.17.RELEASE | HIGH | 3 | Highest | 28


Does the Jenkins job that runs the owasp checker produce any logs that indicate whether it is using the owasp cli or the mvn plugin? Can you tell if the Jenkins plugin is configuring the dependency checker with any particular set of flags?

With regards to the above vulnerabilities, we will be publishing a patch release to bump the dependency versions for spring and jackson-databind dependencies.

**Comment by hannes-angst** [comment]
I have no longer access to the machine, but I noticed NVD updates on the CVE where made recently.

https://nvd.nist.gov/vuln/detail/CVE-2016-0264/change-record?changeRecordedOn=6%2f24%2f2019+10%3a07%3a03+AM&type=new#0

Same goes for the CVE-2014-8892. 

This either means that the false positives got fixed at the source or that the owasp tool behaves differently.

Feel free to close this ticket or investigate further. :)

**Comment by rtveitch** [comment]
Cheers! I'll leave the ticket open for now and close once we get the next patch release out with the aforementioned dependency updates, thanks!

**Comment by rtveitch** [comment]
Closing this ticket as we released `1.0.2` a few days ago which addresses several outstanding CVEs. We'll continue to publish additional patch releases through time to address new CVEs.

https://github.com/IBM/cos-spring/releases/tag/1.0.2
